File: src/test/java/PlayersTest.java
import de.floribe2000.warships_java.direct.account.*;
import de.floribe2000.warships_java.direct.api.ApiBuilder;
import de.floribe2000.warships_java.direct.api.Region;

import java.io.FileInputStream;
import java.io.IOException;
import java.util.Properties;

import de.floribe2000.warships_java.direct.warships.Statistics;
import de.floribe2000.warships_java.direct.warships.StatisticsRequest;
import org.junit.Test;

public class PlayersTest {

    private String apiKey;

    private final String instanceName = "TEST";

    public PlayersTest() throws IOException {
        Properties PROPERTIES = new Properties();
        PROPERTIES.load(new FileInputStream("Warships.properties"));
        apiKey = PROPERTIES.getProperty("APIKEY");
    }

    @Test
    public void testPlayersRequest() {
        ApiBuilder.createInstance(apiKey, instanceName);
        PlayersRequest request = PlayersRequest.createRequest().region(Region.EU).searchText("floribe");
        Players result = request.fetch();
        assert result.getStatus().get() : result;
    }

    @Test
    public void testPlayerPersonalDataRequest() {
        //TODO
        ApiBuilder.createInstance(apiKey, instanceName);
        PlayersPersonalDataFull result = PlayersPersonalDataFullRequest.createRequest().region(Region.EU).addAccountId(537376379)
                .addExtraField(PlayersPersonalDataFullRequest.ExtraField.PVE)
                .addExtraField(PlayersPersonalDataFullRequest.ExtraField.RANK_SOLO).fetch();
        assert result.getStatus().get() : result;
    }

    @Test
    public void testPlayersAchievments() {
        ApiBuilder.createInstance(apiKey, instanceName);
        PlayersAchievments result = PlayersAchievmentsRequest.createRequest().region(Region.EU).accountId(537376379).fetch();
        assert result.getStatus().get() : result;
        assert PlayersAchievmentsRequest.AchievmentElement.retrieveElement("Solo Warrior") != null;
    }

    @Test
    public void testPlayersStatisticsByDate() {
        ApiBuilder.createInstance(apiKey, instanceName);
        PlayersStatisticsByDate result1 = PlayerStatisticsByDateRequest.createRequest().region(Region.EU).accountId(537376379).addDate("20200318").fetch();
        assert result1.getStatus().get() : result1;
        PlayersStatisticsByDate result = PlayerStatisticsByDateRequest.createRequest().region(Region.EU).accountId(537376379).addDate("20200228").addDate("20200118")
                .extra(PlayerStatisticsByDateRequest.ExtraField.PVE).fetch();
        assert result.getStatus().get() : result;
    }

    @Test
    public void testPlayersWarshipsStatistics() {
        ApiBuilder.createInstance(apiKey, instanceName);
        Statistics result = StatisticsRequest.createRequest().region(Region.EU).accountId(537376379).addExtraField(StatisticsRequest.ExtraField.PVE).fetch();
        assert result.getStatus().get() : result;
    }

    @Test
    public void testInstanceCount() {
        assert ApiBuilder.getInstanceSize() == 1 : ApiBuilder.getInstanceSize();
    }

}

File: src/test/java/PlayersTest.kt
import de.floribe2000.warships_java.direct.account.*
import de.floribe2000.warships_java.direct.api.ApiBuilder
import de.floribe2000.warships_java.direct.account.PlayersRequest
import de.floribe2000.warships_java.direct.api.typeDefinitions.Region
import de.floribe2000.warships_java.direct.warships.StatisticsRequest
import org.junit.jupiter.api.Test
import java.lang.InterruptedException
import java.util.Properties
import java.io.FileInputStream

class PlayersTest {
    private val apiKey: String
    private val instanceName = "TEST"

    @Test
    fun testPlayersRequest() {
        ApiBuilder.createInstance(apiKey, instanceName)
        val request = PlayersRequest.createRequest().region(Region.EU).searchText("floribe")
        val result = request.fetch()
        assert(result.status.get()) { result }
    }

    @Test
    fun testPlayerPersonalDataRequest() {
        //TODO
        ApiBuilder.createInstance(apiKey, instanceName)
        val result = PlayersPersonalDataFullRequest.createRequest().region(Region.EU).addAccountId(537376379)
                .addExtraField(PlayersPersonalDataFullRequest.ExtraField.PVE)
                .addExtraField(PlayersPersonalDataFullRequest.ExtraField.RANK_SOLO).fetch()
        assert(result.status.get()) { result }
    }

    @Test
    fun testPlayersAchievements() {
        ApiBuilder.createInstance(apiKey, instanceName)
        val result = PlayersAchievmentsRequest.createRequest().region(Region.EU).addAccountId(537376379).fetch()
        assert(result.status.get()) { result }
    }

    @Test
    fun testPlayersStatisticsByDate() {
        ApiBuilder.createInstance(apiKey, instanceName)
        val result1 = PlayerStatisticsByDateRequest.createRequest().region(Region.EU).accountId(537376379).addDate("20200318").fetch()
        assert(result1.status.get()) { result1 }
        val result = PlayerStatisticsByDateRequest.createRequest().region(Region.EU).accountId(537376379).addDate("20200228").addDate("20200118")
                .extra(PlayerStatisticsByDateRequest.ExtraField.PVE).fetch()
        assert(result.status.get()) { result }
    }

    @Test
    fun testPlayersWarshipsStatistics() {
        ApiBuilder.createInstance(apiKey, instanceName)
        val result = StatisticsRequest.createRequest().region(Region.EU).accountId(537376379).addExtraField(StatisticsRequest.ExtraField.PVE).fetch()
        assert(result.status.get()) { result }
    }

    @Test
    fun testAsyncPlayersRequest() {
        ApiBuilder.createInstance(apiKey, instanceName)
        val name = "floribe2000"
        PlayersRequest.createRequest().region(Region.EU).searchText(name).fetchAsync { result: Players ->
            assert(result.status.get()) { result }
            assert(result.data!![0].nickname == name) { result }
        }
        try {
            Thread.sleep(3000)
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }
    }

    @Test
    fun testAsyncPlayersRequestModified() {
        ApiBuilder.createInstance(apiKey, instanceName)
        val name = "floribe2000"
        val name2 = "MrDios"
        val request = PlayersRequest.createRequest().region(Region.EU).searchText(name)
        request.fetchAsync { result: Players ->
            try {
                Thread.sleep(500)
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }
            println(result)
            assert(result.status.get()) { result }
            assert(result.data!![0].nickname == name) { result }
        }
        val players = request.searchText(name2).fetch()
        println(players)
        assert(players.status.get()) { players }
        assert(players.data!![0].nickname!!.contains(name2)) { players }
        try {
            Thread.sleep(3000)
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }
    }

    init {
        val properties = Properties()
        properties.load(FileInputStream("Warships.properties"))
        apiKey = properties.getProperty("APIKEY")
    }
}